yt_dlp/extractor/indavideo.py

```diff
@@ -78,13 +73,12 @@ def _real_extract(self, url):
         for video_url in video_urls:
             height = int_or_none(self._search_regex(
                 r'\.(\d{3,4})\.mp4(?:\?|$)', video_url, 'height', default=None))
-            if filesh:
-                if not height:
-                    continue
-                token = filesh.get(compat_str(height))
-                if token is None:
-                    continue
-                video_url = update_url_query(video_url, {'token': token})
+            if not height and len(filesh) == 1:
+                height = int(list(filesh.keys())[0])
+            token = filesh.get(compat_str(height))
```

Comment:
```suggestion
            token = filesh.get(str(height))
```

can remove the compat_str now